example_client/multi_inputs.py

```diff
@@ -0,0 +1,279 @@
+def parse_output(thr_id, res, frame):
+  for batch, data in enumerate(res):
+    pred = data[0]
+    for values in enumerate(pred):
+      # tuple
+      index = values[0]
+      l_pred = values[1]
+
+      # actual predictions
+      img_id = l_pred[0]
+      label = l_pred[1]
+      conf = l_pred[2]
+      x_min = l_pred[3]
+      y_min = l_pred[4]
+      x_max = l_pred[5]
+      y_max = l_pred[6]
+
+      # preventing any wrong array indexing (for RMNet)
+      if label >= 4:
+        log.error('Unsupported class detected in camera {}'.format(thr_id))
+        label = 4
+
+      # Do you want confidence level to be passed from command line?
+      if img_id != -1 and conf >= 0.75:
+        # draw the bounding boxes on the frame
+        height, width = frame.shape[:2]
+        cv2.rectangle(frame, (int(width * x_min), int(height * y_min)),
+              (int(width * x_max), int(height * y_max)), COLORS[int(label)], 2)
+        cv2.putText(frame, str(CLASSES[int(label)]), (int(width * x_min)-10,
+              int(height * y_min)-10), cv2.FONT_HERSHEY_SIMPLEX, 0.5,
+              COLORS[int(label)], 2)
+  return frame
+
+# This is common for both the camera & video files
+def thread_function(thr_id, network_name, input_layer, output_layer, input_dimension,
+                    stub, thr_lock, disp_buf, src_type, src_name):
+  if src_type == "Camera":
+    # UVC camera init - camera threads always come first and we use it
+    # to generate the camera indexes
+    cam = cv2.VideoCapture(thr_id)
+    if not (cam.isOpened()):
+      log.error("Failed to open the UVC camera {}".format(thr_id))
+      return
+
+    cam.set(cv2.CAP_PROP_FRAME_WIDTH, cam_width)
+    cam.set(cv2.CAP_PROP_FRAME_HEIGHT, cam_height)
+    # not all UVC cameras honor below request
+    cam.set(cv2.CAP_PROP_FPS, cam_fps)
+    # If your camera sends other than MJPEG, change below
+    cam.set(cv2.CAP_PROP_FOURCC, cv2.VideoWriter_fourcc(*"MJPG"))
+  elif src_type == "Video":
+    # Assumption: src_name will be valid
+    cam = cv2.VideoCapture(src_name)
+
+  # stats
+  fps = 0                   # camera fps
+  inf_fps = 0               # inference fps
+  dropped_fps = 0           # dropped frame fps
+  cam_start_time = time()
+
+  request = predict_pb2.PredictRequest()
+  # Note: Pls maintain the same name while launching ovms docker container
+  request.model_spec.name = network_name
+
+  global exit_ok
+  while exit_ok == False:
+    ret, frame = cam.read()
+
+    if src_type == "Video":
+      # restart the video file when it reaches the end
+      if not ret:
+        cam.set(cv2.CAP_PROP_POS_FRAMES, 0)
+        continue
+      # normalize the video frame dimension to that of the camera
+      else:
+        # to maintain the frame inferencing parity with the cameras, lets sleep
+        # here to maintain cam_fps speed
+        sleep((1000 / cam_fps) / 1000)
+        frame = cv2.resize(frame, (cam_width, cam_height))
```

Comment:
Why do you resize here to `(cam_width, cam_height)` and few lines below to `(input_dimension, input_dimension) `?
Is it necessary to do both?

Reply:
This is to resize the video frames read from the files.